base/abstractarraymath.jl

```diff
@@ -92,6 +92,10 @@ imag{T<:Real}(x::AbstractArray{T}) = zero(x)
 +{T<:Number}(x::AbstractArray{T}) = x
 *{T<:Number}(x::AbstractArray{T,2}) = x
 
+function lerpi(j::Integer, d::Integer, A::AbstractArray, B::AbstractArray)
```

Comment:
and `r.start` and `r.stop` can be either scalars or arrays? since this method is just `lerpi.(j, d, A, B)`, it would make more sense in my head to move the dispatch to the call site, if `unsafe_getindex` needs to still call non-broadcast `lerpi` on scalars for performance, then couldn't this dot call be moved to a method of `unsafe_getindex` for array inputs rather than a method of `lerpi` ?